### ISSUE TITLE
Unsupport type from raw stream while piping .zsh_history

### ISSUE THREAD

**Issue by alfa07** [nushell/nushell#6880]
### Describe the bug

```
❯ : open ~/.zsh_history | lines
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #17:1:1]
 1 │ open ~/.zsh_history | lines
   · ──┬─
   ·   ╰── Unsupport type from raw stream
   ╰────

```

### How to reproduce

1. Run `open ~/.zsh_history | lines`

### Expected behavior

Expected to produce lines from the history file, e.g. `open $nu.history-path | lines` works fine.

### Configuration

❯ : version | transpose key value | to md --pretty
| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.70.0                                          |
| branch             |                                                 |
| commit_hash        |                                                 |
| build_os           | macos-aarch64                                   |
| build_target       | aarch64-apple-darwin                            |
| rust_version       | rustc 1.64.0                                    |
| cargo_version      | cargo 1.64.0 (387270bc7 2022-09-16)             |
| pkg_version        | 0.70.0                                          |
| build_time         | 2022-10-18 18:13:36 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

### Additional context

I am trying to port my zsh history into nushell.

**Comment by alfa07** [comment]
I suspect that support for binary is required here: https://github.com/nushell/nushell/blob/4858a9a81787df53f9f3840f04abfd36c94599eb/crates/nu-command/src/filters/lines.rs#L224

**Comment by sholderbach** [comment]
Yeah trying to `String` should fall back to the raw bytes if that fails (or if desired use the unicode replacement from `std::String::from_bytes` to mask the bad bytes)

**Comment by JohnSwiftC** [comment]
Is this still open to be completed? I read the first PR and the response, and I would implement that warning if it's still wanted.

**Comment by sholderbach** [comment]
First thing would be to check what actually happens now when `lines` receives non-UTF8 input. There has been a lot of work since this issue was last looked at.

**Comment by ravindar-10** [comment]
Hi @alfa07 ,
if it's still open,Can I start working on this issue?

**Comment by eyupcanakman** [comment]
PR #17810 used `String::from_utf8_lossy`, which sholderbach listed as one of two options in https://github.com/nushell/nushell/issues/6880#issuecomment-1288861436, but @cptpiepmatz closed it pointing back at that same comment. Lossy replacement or raw-byte fallback? I'll check what `lines` does on current main first, since a lot has changed since 2022.
